### ISSUE TITLE
We need to actually fix our UpdateWhileInUse manifest (was: Terminal closes itself unexpectedly from time to time)

### ISSUE THREAD

**Issue by ygoe** [microsoft/terminal#18119]
### Windows Terminal version

1.21.2911.0

### Windows build number

10.0.22631.3880

### Steps to reproduce

1. Start the terminal app
2. Let it run with one or two tabs open running console applications

### Expected Behavior

Terminal should stay open until I close it.

### Actual Behavior

Terminal closes itself from time to time, also stopping all running console applications. Last time was last night at 2024-10-28 01:50 +0100. I have no event log of this but it could be once in a few months. All other applications are still running, nothing else stops unexpectedly.

**Comment by ygoe** [comment]
If it's an auto-update situation and the Store app force-closes the Terminal, then there's a serious problem here. The Store must not have the power to randomly kill other apps whenever it wants. If that's what it does, I'll have to force-disable those updates. I expect it to be at least so cooperative and inform me that it wants to install an update and requires the Terminal to close. If I can't afford to stop all currently running console apps (which might do long-running work), I must have the ability to defer this for arbitrary time.

I'd appreciate it doing its updates when I also let Windows Update run. Then I'm aware that a reboot is needed and all apps are closed, so I'll only do that when possible. IIRC, PowerShell (the new one) is also only updated along with Windows Updates, so Terminal should do just the same.

**Comment by zadjii-msft** [comment]
Alas, this actually came up in a mail thread while I was out. For my own reference: "**Subject**: RE: UpdateWhileInUse issue follow-up"

tldr of that thread is that we didn't update our manifest entirely correctly. Sounds like just adding the new property () wasn't good enough. We'll also need to set our `MaxVersionTested` to `10.0.26100.0`.

This alas will have other, unexpected consequences (apparently), so we will need to do quite a bit of validation on our side.

**Comment by ygoe** [comment]
Good to hear that you know what's happening and what might fix it.

Until this is fixed, can I selectively prevent Terminal from being updated automatically? Without disabling all app updates altogether? Or can I only shut down the whole thing all at once? I remember Android has auto-update settings per app (not that auto-update would actually reliably do anything there, but at least the option is there). Couldn't find that in Windows or the Store app now.

**Comment by hammypants** [comment]
Still happening.

> Until this is fixed, can I selectively prevent Terminal from being updated automatically? Without disabling all app updates altogether? Or can I only shut down the whole thing all at once? I remember Android has auto-update settings per app (not that auto-update would actually reliably do anything there, but at least the option is there). Couldn't find that in Windows or the Store app now.

I would also like to know this.

**Comment by ygoe** [comment]
No info yet. I've disabled the automatic updates in Windows Store. Like I already did for Windows altogether. Considering Microsoft's ever-decreasing updates quality, extra carefulness is due. Never let it update when it wants to interrupt your work. Do it shortly before new updates appear. By then most issues should be sorted out.

**Comment by zadjii-msft** [comment]
Alas, not that I'm aware of. This has been one of rougher edges with the Store as long as I can remember. Updating our manifest is the only way I know how to fix this. Just a bummer that the magic keyword here needed a _second_ magic keyword that we didn't know about, and has other possibly unexpected consequences 😕
